### PR TITLE
LPS-143051 Already added items can be added to a collection variation again

### PR DESCRIPTION
**Pull request by BarbaraCabrera**

### CODE REVIEW

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by ealonso** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/service/impl/AssetListEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -114,7 +114,22 @@ public void addAssetEntrySelections(
 					segmentsEntryId, StringPool.BLANK, serviceContext);
 		}
 
+		List<AssetListEntryAssetEntryRel> assetListEntryAssetEntryRels =
+			_assetListEntryAssetEntryRelLocalService.
+				getAssetListEntryAssetEntryRels(
+					assetListEntryId, QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
 		for (long assetEntryId : assetEntryIds) {
+			boolean exist = ListUtil.exists(
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/service/impl/AssetListEntryLocalServiceImpl.java:123]: no need to define the variable, use directly the expression in the clause

File: modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/service/impl/AssetListEntryLocalServiceImpl.java [hunk 2]
```diff
@@ -114,7 +114,22 @@ public void addAssetEntrySelections(
...
+				assetListEntryAssetEntryRels,
+				assetListEntryAssetEntryRel ->
+					assetListEntryAssetEntryRel.getAssetEntryId() ==
+						assetEntryId);
+
+			if (exist) {
+				continue;
+			}
+
 			_assetListEntryAssetEntryRelLocalService.
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-service/src/main/java/com/liferay/asset/list/service/impl/AssetListEntryLocalServiceImpl.java:133]: add the just created AassetListEntryAssetEntryRel to assetListEntryAssetEntryRels, in this way we can ensure that relations won't be duplicated although assetEntryIds contains duplicated ids

**Review by lfbesada — CHANGES_REQUESTED** [review]
@BarbaraCabrera great job! 👏 
Could you please add an integration test to ensure the behavior?
Thanks!

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:relevant

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:relevant

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java [hunk 3]
```diff
@@ -104,6 +105,78 @@ public void testAddDuplicateAssetListEntry() throws PortalException {
...
+		AssetListEntry assetListEntry =
+			_assetListEntryService.addAssetListEntry(
+				_group.getGroupId(), RandomTestUtil.randomString(),
+				AssetListEntryTypeConstants.TYPE_MANUAL, serviceContext);
+
+		AssetEntry assetEntry1 = AssetTestUtil.addAssetEntry(
+			_group.getGroupId(), null,
+			TestAssetRendererFactory.class.getName());
+
+		// array de 1
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java:125]: I am sorry, don't add comments

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java [hunk 4]
```diff
@@ -104,6 +105,78 @@ public void testAddDuplicateAssetListEntry() throws PortalException {
...
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId());
+
+		AssetListEntry assetListEntry =
+			_assetListEntryService.addAssetListEntry(
+				_group.getGroupId(), RandomTestUtil.randomString(),
+				AssetListEntryTypeConstants.TYPE_MANUAL, serviceContext);
+
+		AssetEntry assetEntry1 = AssetTestUtil.addAssetEntry(
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java:121]: There is only one, its name should be assetEntry

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java [hunk 5]
```diff
@@ -104,6 +105,78 @@ public void testAddDuplicateAssetListEntry() throws PortalException {
...
+		int assetListEntriesAssetEntryRelsCount =
+			_assetListEntryAssetEntryRelLocalService.
+				getAssetListEntryAssetEntryRelsCount(
+					assetListEntry.getAssetListEntryId());
+
+		_assetListEntryService.addAssetEntrySelection(
+			assetListEntry.getAssetListEntryId(), assetEntry1.getEntryId(), 0,
+			serviceContext);
+
+		int assetListEntriesAssetEntryRelsCountAfterDoubleInsertion =
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java:140]: Don't need to declare var, use it inline

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java [hunk 6]
```diff
@@ -104,6 +105,78 @@ public void testAddDuplicateAssetListEntry() throws PortalException {
...
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(
+				_group.getGroupId(), TestPropsValues.getUserId());
+
+		AssetListEntry assetListEntry =
+			_assetListEntryService.addAssetListEntry(
+				_group.getGroupId(), RandomTestUtil.randomString(),
+				AssetListEntryTypeConstants.TYPE_MANUAL, serviceContext);
+
+		AssetEntry assetEntry1 = AssetTestUtil.addAssetEntry(
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java:163]: Same as above: there is only one, its name should be assetEntry

File: modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java [hunk 7]
```diff
@@ -104,6 +105,78 @@ public void testAddDuplicateAssetListEntry() throws PortalException {
...
+			_group.getGroupId(), null,
+			TestAssetRendererFactory.class.getName());
+
+		_assetListEntryService.addAssetEntrySelections(
+			assetListEntry.getAssetListEntryId(),
+			new long[] {assetEntry1.getEntryId(), assetEntry1.getEntryId()}, 0,
+			serviceContext);
+
+		int assetListEntriesAssetEntryRelsCount =
```
Comment (lfbesada) [inline modules/apps/asset/asset-list-test/src/testIntegration/java/com/liferay/asset/list/service/test/AssetListEntryServiceTest.java:172]: Same as above: don't need to declare var, use it inline

**Review by lfbesada — CHANGES_REQUESTED** [review]
Good job! 🎉 Just a minor source formatting issues

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:relevant

**Review by lfbesada — APPROVED** [review]
💃

**Comment by lfbesada** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward

**Comment by lfbesada** [comment]
ci:forward

**Pull request closed without merging**